core/src/main/java/tc/oc/pgm/kits/KitParser.java

```diff
@@ -422,7 +429,12 @@ public ItemStack parseItem(Element el, Material type, short damage) throws Inval
       throw new InvalidXMLException("Invalid item/block", el);
     }
 
+    if (amount == -1 && !itemStack.getType().isBlock()) {
+      throw new InvalidXMLException("infinity can only be applied to a block material", el);
```

Comment:
Why can't we support infinite snowballs or enderpearls?

Reply:
I haven't tested them at the time, I'll see if works.